### PR TITLE
feat : Add login/signup

### LINKED ISSUE #5: Add : Authentication

### PR DESCRIPTION
**Pull request by ZuberDunge**
image

### CODE REVIEW

**Review by prachi-sahani — COMMENTED** [review]
overall the code is clean and easy to understand
liked the implementation for remember me feature
good to merge

File: src/context/authContext.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import { getAuth } from "firebase/auth";
+
+import {
+    createUserWithEmailAndPassword,
+    signInWithEmailAndPassword,
+    onAuthStateChanged,
+    signOut,
+} from "firebase/auth";
+import { auth } from "./../config/firebase-config";
+import { db } from "./../config/firebase-config";
```
Comment (abhij1607) [inline src/context/authContext.js:20]: can be imported at the same line

File: src/context/authContext.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    });
+
+    const { setIsLoggedIn } = useContext(ListContext)
+    const [stopLoading, setStopLoading] = useState(false)
+    const registerUser = async (userName, registerEmail, registerPassword) => {
+        try {
+            const { user } = await createUserWithEmailAndPassword(auth, registerEmail, registerPassword)
+            localStorage.setItem("userID", JSON.stringify(user.uid))
+            await setDoc(doc(db, user.uid, user.uid), { title: "", img: "", link: "", username: userName })
+            console.log('Document Added')
```
Comment (abhij1607) [inline src/context/authContext.js:49]: remove console.log

File: src/context/authContext.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            setIsLoggedIn(true)
+            setStopLoading(true)
+            // remember me
+            if (rememberMe) {
+                localStorage.setItem("userEmail", registerEmail)
+                localStorage.setItem("userPassword", registerPassword)
+            }
+
+        } catch (error) {
+            console.log(error);
```
Comment (abhij1607) [inline src/context/authContext.js:59]: remove console.log, error could be displayed to user via any function

File: src/context/authContext.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            userExistFun()
+        }
+
+    }
+
+    useEffect(() => {
+        if (localStorage.getItem("userEmail") != null) {
+            setLoginEmail(localStorage.getItem("userEmail"))
+            console.log(localStorage.getItem("userEmail") != null);
```
Comment (abhij1607) [inline src/context/authContext.js:68]: remove console.log

File: src/context/authContext.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    }, [])
+    const login = async (email, password) => {
+        try {
+            const { user } = await signInWithEmailAndPassword(
+                auth,
+                email,
+                password
+            );
+            localStorage.setItem("userID", JSON.stringify(user.uid))
+            console.log(user);
```
Comment (abhij1607) [inline src/context/authContext.js:82]: remove console.log

File: src/context/authContext.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            setIsLoggedIn(true)
+
+            // remember me
+            if (rememberMe) {
+                localStorage.setItem("userEmail", loginEmail)
+                localStorage.setItem("userPassword", loginPassword)
+            }
+
+        } catch (error) {
+            console.log(error);
```
Comment (abhij1607) [inline src/context/authContext.js:92]: remove console.log

File: src/context/authContext.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            wrongPasswordFun()
+        }
+    };
+
+    const logout = async () => {
+        await signOut(auth);
+        console.log("logged out");
```
Comment (abhij1607) [inline src/context/authContext.js:99]: remove console.log

File: src/context/authContext.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        localStorage.removeItem("userID")
+        setIsLoggedIn(false)
+    };
+
+
+    const wrongPasswordFun = () => {
+        setTimeout(() => {
+            setwrongPassword(false)
+        }, 2000)
+        console.log("settime");
```
Comment (abhij1607) [inline src/context/authContext.js:109]: remove console.log

File: src/context/authContext.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        setwrongPassword(true)
+    }
+
+    const userExistFun = () => {
+        setTimeout(() => {
+            setuserExist(false)
+        }, 2000)
+        console.log("settime");
```
Comment (abhij1607) [inline src/context/authContext.js:117]: remove console.log

File: src/user/login.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    } = useForm();
+
+    const signUpUser = (data) => {
+        registerUser(data.userName, data.registerEmail, data.registerPassword)
+    };
+
+    const signInUser = (data) => {
+        login(data.userEmail, data.userPassword)
+    };
+    console.log(user, !user);
```
Comment (abhij1607) [inline src/user/login.js:24]: remove console.log

**Review by abhij1607 — COMMENTED** [review]
Very neat & readable code | | Good naming conventions used

left few suggestions in comments

File: src/user/login.js (hunk 10 above)
Comment (ZuberDunge) [reply src/user/login.js:24]: thank you for the review Abhi, I'll make the changes.

**Pull request merged**